cpp/lib/answer/json.cpp

```diff
@@ -321,6 +327,56 @@ JSON JSONBuilder::buildArray(const nlohmann::json& from) {
     return JSON(std::move(a));
 }
 
+void JSON::appendToString(std::stringstream& ss, const JSON& json) const {
+    switch (json.type()) {
+        case JSONType::STRING: {
+            ss << "\"" << json.asString() << "\"";
+            break;
+        }
+        case JSONType::ARRAY:{
+            ss << "[";
+            for (size_t i = 0; i < json.asArray().size(); ++i) {
+                if (i > 0) ss << ", ";
+                appendToString(ss, json.asArray()[i]);
+            }
+            ss << "]";
+            break;
+        }
+        case JSONType::MAP: {
+            ss << "{";
+            bool first = true;
+            for (const auto& pair : json.asMap()) {
+                if (!first) ss << ", ";
+                ss << "\"" << pair.first << "\": ";
+                appendToString(ss, pair.second);
+                first = false;
+            }
+            ss << "}";
+            break;
+        }
+        case JSONType::BOOLEAN: {
+            ss << (json.asBoolean() ? "true" : "false");
+            break;
+        }
+        case JSONType::LONG: {
+            ss << json.asLong();
+            break;
+        }
+        case JSONType::DOUBLE: {
+            ss << json.asDouble();
+            break;
+        }
+        case JSONType::NULL_VALUE: {
+            ss << "null";
+            break;
+        }
+        case JSONType::INVALID: {
+            throw Utils::exception(DriverError::INVALID_JSON_CAST, NAME(_type), NAME(JSONType::STRING));
```

Comment:
This is misleading. We're not casting to a JSON string here, but serialising.

Also, this references `this->_type` rather than `json.type()`.

Reply:
`INVALID_NATIVE_HANDLE`